### DISCUSSION TITLE
Nova updates blocks meta/favicon override

### DISCUSSION THREAD

**Discussion by jarrodparkes** [laravel/nova-issues#3962] [Ideas & Feature Requests]
- Laravel Version: 9.7.0
- Nova Version: 4.1.0
- PHP Version: 8.0.2
- Database Driver & Version: PSQL, 12.2
- Operating System and Version: macOS Monterey, 12.3
- Browser type and version: Safari, 15.4

### Description:

When #1346 was fixed, it allowed for the meta partial to be used to add favicon (among other things) into Nova meta. It appears that has been removed.

### Detailed steps to reproduce the issue on a fresh Nova installation:

- Edit `/resources/views/vendor/nova/partials/meta.blade.php` to include a favicon
- Favicon doesn't work

**Comment by sebsel** [comment]
I miss this. The workaround might seem like just adding a `/favicon.ico`, but we actually used this to give the favicon multiple colors depending on the app env... makes it easy to find an open production or staging tab in your browser.

**Reply by sebsel** [reply]
Maybe don't do this but this works.

add to composer.json:
```json
        "post-install-cmd": [
            "@php artisan hack:patch-nova-meta"
        ],
        "post-update-cmd": [
            ...
            "@php artisan hack:patch-nova-meta --warn"
        ]
```

`app\Console\Commands\PatchNovaMeta.php`
```php
<?php

namespace App\Console\Commands;

use Illuminate\Console\Command;

class PatchNovaMeta extends Command
{
    /**
     * The name and signature of the console command.
     */
    protected $signature = 'hack:patch-nova-meta {--warn}';

    /**
     * The console command description.
     */
    protected $description = "Include our own meta tags because Nova doesn't";

    /**
     * Execute the console command.
     */
    public function handle()
    {
        // HACK but hopefully a responsible one

        if ($this->option('warn')) {
            if ($this->confirm('Does the new version of Nova have support for a meta tag?')) {
                $this->warn(
                    'Please remove '.__FILE__.' and the `post-install-cmd` and '.
                    '`post-update-cmd` hooks in the composer.json'
                );
                return 0;

            } else {
                $this->info('Gotta wait until they change it then.');
            }
        }

        $layoutFile = base_path('/vendor/laravel/nova/resources/views/layout.blade.php');
        $layout = file_get_contents($layoutFile);

        $needle = '<meta charset="utf-8">';
        $addition = ' @include("vendor.nova.partials.meta")';

        if (! str_contains($layout, $addition)) {
            $layout = str_replace($needle, $needle.$addition, $layout);
            file_put_contents($layoutFile, $layout);
            $this->info('Patched the Nova layout to include our favicons');
        } else {
            $this->info('Your Nova install already had favicons');
        }

        return 0;
    }
}
```

**Reply by Milkhan** [reply]
@davidhemphill I belive this is no longer an issue right? Just create a meta.blade.php with a favicon in resources/views/vendor/nova/partials/, works for me.

**Reply by sebsel** [reply]
I believe this was fixed indeed.

**Reply by Krlos891** [reply]
@Milkhan Thanks for that. I was looking for this so many time!

**Reply by sn-seanh** [reply]
Hey Guys,

I am not coming right with this one, Can anyone paste an example of their meta.blade.php here?

**Reply by sn-seanh** [reply]
Nvm, I came right.

favicon in public folder and then `<link rel="icon" type="image/x-icon" href="{{ asset('favicon.ico') }}" >` in /resources/views/vendor/nova/partials/meta.blade.php
